datafusion/functions/src/math/log.rs

```diff
@@ -72,37 +72,28 @@ impl Default for LogFunc {
 
 impl LogFunc {
     pub fn new() -> Self {
+        // Converts decimals & integers to float64, accepting other floats as is
```

Comment:
Main fix here for signature; to me this is more readable than the previous signature, and importantly it accepts any decimals regardless of precision/scale.

Note to self: add tests for decimals of different precision/scale if they don't exist